lib/engine/game/g_1822.rb

```diff
@@ -62,13 +78,94 @@ def operating_round(round_num)
         ], round_num: round_num)
       end
 
+      def setup
+        @bidding_token_per_player = init_bidding_token
+        setup_companies
+        setup_bidboxes
+      end
+
+      def sorted_corporations
+        @corporations.select { |c| c.floated? && c.type == :major }
+      end
+
+      def stock_round
+        Round::G1822::Stock.new(self, [
+          Step::DiscardTrain,
+          Step::Exchange,
+          Step::SpecialTrack,
+          Step::G1822::BuySellParShares,
+        ])
+      end
+
+      def bidbox_minors
+        @companies.select { |c| c.id[0] == 'M' && (c.owner.nil? || c.owner == @bank) }.map.first(4)
```

Comment:
what's this map here for?

don't call owner.nil?

&& !c.owner || c.owner == @bank

Reply:
Just a lack of knowledge of ruby, i probably saw code something like it and did it about the same way.